header.js

```diff
@@ -20,7 +22,12 @@ const BN = utils.BN
    * @prop {Buffer} extraData
    * @prop {Array.<Buffer>} raw an array of buffers containing the raw blocks.
    */
-var BlockHeader = module.exports = function (data) {
+var BlockHeader = module.exports = function (data, opts) {
+  opts = opts || {}
+  this._chain = opts.chain ? opts.chain : 'mainnet'
+  this._hardfork = 'byzantium'
+  this._common = new Common(this._chain, this._hardfork)
```

Comment:
There are several tests and other examples where Block.number is set directly. Shouldn't this._hardfork and this._common be automatically updated to stay in sync with the number? Perhaps you can override the setter for BlockHeader.number so that whenever the number is changed, this._common and this._hardfork are updated to ensure they are in sync with the number.